From d9a7a4c1dde51c6764c35ce639b02fd939e076cc Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:32:54 -0400
Subject: [alsa] hda: Disable AA-mix on Sony Vaio S13

Message-id: <140972823573-304-patchset-jkysela@redhat.com>
Patchwork-id: 91000
O-Subject: [RHEL 7.1 PATCH 303/442] ALSA: hda - Disable AA-mix on Sony Vaio S13
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

The analog-loopback causes the speaker noises even if it's set to zero
volume.  As a simple workaround, just get rid of the loopback mixer.

Bugzilla: https://bugzilla.novell.com/show_bug.cgi?id=873704
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: e9bd7d5ce8d14517aa1383bb4f84e55e7c0d4363)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_realtek.c b/sound/pci/hda/patch_realtek.c
index da2923d..1f06007 100644
--- a/sound/pci/hda/patch_realtek.c
+++ b/sound/pci/hda/patch_realtek.c
@@ -4196,6 +4196,7 @@ enum {
  ALC269_FIXUP_ASUS_G73JW,
  ALC269_FIXUP_LENOVO_EAPD,
  ALC275_FIXUP_SONY_HWEQ,
+ ALC275_FIXUP_SONY_DISABLE_AAMIX,
  ALC271_FIXUP_DMIC,
  ALC269_FIXUP_PCM_44K,
  ALC269_FIXUP_STEREO_DMIC,
@@ -4301,6 +4302,12 @@ static const struct hda_fixup alc269_fixups[] = {
   .chained = true,
   .chain_id = ALC275_FIXUP_SONY_VAIO_GPIO2
  },
+ [ALC275_FIXUP_SONY_DISABLE_AAMIX] = {
+  .type = HDA_FIXUP_FUNC,
+  .v.func = alc_fixup_disable_aamix,
+  .chained = true,
+  .chain_id = ALC269_FIXUP_SONY_VAIO
+ },
  [ALC271_FIXUP_DMIC] = {
   .type = HDA_FIXUP_FUNC,
   .v.func = alc271_fixup_dmic,
@@ -4814,6 +4821,7 @@ static const struct snd_pci_quirk alc269_fixup_tbl[] = {
  SND_PCI_QUIRK(0x104d, 0x9073, "Sony VAIO", ALC275_FIXUP_SONY_VAIO_GPIO2),
  SND_PCI_QUIRK(0x104d, 0x907b, "Sony VAIO", ALC275_FIXUP_SONY_HWEQ),
  SND_PCI_QUIRK(0x104d, 0x9084, "Sony VAIO", ALC275_FIXUP_SONY_HWEQ),
+ SND_PCI_QUIRK(0x104d, 0x9099, "Sony VAIO S13", ALC275_FIXUP_SONY_DISABLE_AAMIX),
  SND_PCI_QUIRK_VENDOR(0x104d, "Sony VAIO", ALC269_FIXUP_SONY_VAIO),
  SND_PCI_QUIRK(0x10cf, 0x1475, "Lifebook", ALC269_FIXUP_LIFEBOOK),
  SND_PCI_QUIRK(0x17aa, 0x20f2, "Thinkpad SL410/510", ALC269_FIXUP_SKU_IGNORE),
-- 
1.7.1